src/dune_rules/melange_rules.ml

```diff
@@ -34,46 +34,71 @@ let local_of_lib ~loc lib =
           (Lib_name.to_string lib_name)
       ]
 
-let js_includes ~loc ~sctx ~target_dir ~requires_link ~scope ~js_ext =
-  let open Resolve.Memo.O in
-  Command.Args.memo
-    (Resolve.Memo.args
-       (let* (libs : Lib.t list) = requires_link in
-        let project = Scope.project scope in
-        let deps_of_lib (lib : Lib.t) =
-          let dst_dir =
-            let lib_dir =
-              let info = local_of_lib ~loc lib |> Lib.Local.info in
-              Lib_info.src_dir info
-            in
-            lib_output_dir ~target_dir ~lib_dir
-          in
-          let open Memo.O in
-          let* source_modules =
-            Dir_contents.modules_of_lib sctx lib
-            >>| Option.value_exn >>| Modules.impl_only
-          in
-          let of_module m =
-            let output = make_js_name ~js_ext ~dst_dir m in
-            Dep.file (Path.build output)
-          in
-          Resolve.Memo.return
-            (List.map source_modules ~f:of_module |> Dep.Set.of_list)
+let virtual_dst_dir_and_modules sctx lib ~loc ~target_dir =
```

Comment:
This function returns now more than `virtual_dst_dir_and_modules` as it includes `vlib` as well.

Not sure if it'd be worth to rename it to something "better" (`vlib_info`?) + define a record for the returned value.

Alternatively, the function could be replaced with smaller functions that do one thing, i.e. `vlib_modules`, `vlib_dst_dir`) and are called separately.

Reply:
I pushed something to simplify this.